### DISCUSSION TITLE
STM32F030F4P6 not working with stm32duino 2.0.0 or later.

### DISCUSSION THREAD

**Discussion by mcer12** [stm32duino/Arduino_Core_STM32#2362] [General]
**Describe the bug**
Tested with blink sketch,  custom board (8mhz crystal + CH340). The chip works with stm32duino 1.9.0 using "STM32F030F4 Demo Board", but when tried with any version past 1.9.0, it uploads the sketch without issues but it doesn't execute properly ie led is not blinking. Other sketches also don't work.

**To Reproduce**
1. Select STM32F030F4 board (8mhz HSE)
2. Upload basic blink sketch to STM32F030F4P6
3. No blinking

**Desktop (please complete the following information):**
 - OS: Windows
 - Arduino IDE version: 1.8.13
 - STM32 core version: 2.7.1
 - Tools menu settings if not the default: No serial
 - Upload method: Serial

**Board (please complete the following information):**
 - Custom STM32F030F4P6 board

**Additional context**
1.9.0 works, 2.0.0+ doesn't work

**Comment by fpistm** [comment]
Converted to discussion.
Chech the clock config it probably changed. Else use the generic using the HSI and redefine the LED.

**Comment by mcer12** [comment]
@fpistm I tried generic also but it doesn't work, I also tried defining the led pin explicitly in case it changed. Will take a look at the clock config.

**Reply by fpistm** [reply]
If generic do not work, then you have to debug. I guess the default clock config is buggy.

**Reply by mcer12** [reply]
@fpistm Upon further investigation it seems like the issue could be with residual code (chip is not erased properly before programming in arduino?) because:
1. SWD programming works perfectly fine
2. Programming previously erased chip (full chip erase in STM32 cube programmer) via Serial also works fine every time
3. Flashing identical sketch in arduino with the one that's onboard also works fine.

**Reply by mcer12** [reply]
Arduino output below:
```
Memory Programming ...
Opening and parsing file: STAIRS_TEST.ino.bin
  File          : STAIRS_TEST.ino.bin
  Size          : 13676 Bytes
  Address       : 0x08000000 


Erasing memory corresponding to segment 0:
Erasing internal memory sectors [0 6]
Download in Progress:


File download complete
Time elapsed during download operation: 00:00:01.900

RUNNING Program ... 
  Address:      : 0x8000000
Start operation achieved successfully
```

Tried it just now with STM32G030K8T6, there's 31 sectors of memory in this chip, but the output shows "Erasing internal memory sectors [0 6]" so most likely that's the issue.

**Reply by fpistm** [reply]
It erase only needed sector for the binary size.

**Reply by fpistm** [reply]
> Flashing identical sketch in arduino with the one that's onboard also works fine.

I don't think it is relevant,  you simply start with the previous binary flashed.
Which version of the ST32cubeProgrammer you used ?

**Reply by mcer12** [reply]
@fpistm ST32cubeProgrammer 2.9.0, I can do full chip erase with SWD or select all sectors and erase them when connected via Serial. After doing this, I can successfully flash it via Serial in Arduino IDE.

**Reply by fpistm** [reply]
What happen if you flash a binary with STM32CubeProgrammer without doing a full erase ?

**Reply by fpistm** [reply]
And please try with the latest version.

**Reply by mcer12** [reply]
@fpistm I tried this few times with the 2.9.0 and it failed just like Arduino IDE. Now after I updated to the latest STM32CubeProgrammer, everything seems to work prerfectly, including arduino ide. Is it possible that the new version of drivers fixed this?

**Reply by fpistm** [reply]
Not link to the driver but to the cubeprogrammer. Probably a bug fixed in recent version.

**Reply by mcer12** [reply]
Thanks for the help, closing the discussion since it works now.
